### DISCUSSION TITLE
Preciso de uma ajuda galera

### DISCUSSION THREAD

**Discussion by Mathues** [community/community#124580] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Opa, galera! Bom dia ou boa noite a todos. Me chamo Matheus, tenho 16 anos e sou apaixonado pela área de TI de várias formas. Meu interesse começou com a montagem de computadores, ou seja, pelo hardware. Com o tempo, também passei a gostar de programação e estudei bastante sobre o assunto.

Porém, muitas vezes me sinto perdido e sem rumo sobre qual caminho seguir ou qual área da programação escolher. Nessas horas, fico desorientado e gostaria de ajuda de alguém com mais experiência e vocação do que eu para me guiar, me ajudar a descobrir mais sobre a área e saber por onde começar. Também seria ótimo encontrar alguém que esteja aprendendo, para compartilharmos experiências e evoluirmos juntos. Se alguém puder me ajudar nessa jornada, eu ficaria muito agradecido.

Obrigado!

**Comment by davevad93** [comment]
Hi @Mathues ,

That depends a lot on what you like, which to me is the most important thing. You've said that you interest began with hardware so you could consider a career in hardware engineering but you have also said that you have started to be interested in programming. The first thing you have to think about is what you really like, that is, if you are more into design oriented web development you should consider learning frontend with HTML, CSS, JavaScript and frameworks like React and Angular. On the other hand, if you are more into the logical part of programming you should focus on backend for example with Python, Java, C#, C++ etcetera and working with databases like SQL. If you like both, you can learn both and specialize in fullstack development. Well, having said that you also have to differentiate between a software developer and a software engineer: software developers design specific computer systems and applications while software engineers work on a larger scale to design, develop, and test entire computer systems and applications. As you can see, there are many options in the IT world and as I said before, in the end you have to focus on what you really enjoy and remember something very important, in this field you never stop learning, I've been doing this for a few years and I constantly learn new things because in the technological world the evolution is constant and you always have to be up to date. I hope that everything I have told you can be of help in choosing your path, good luck!

**Reply by Mathues** [reply]
Oi Daveva93, bom eu gosto da parte de logica sabe de pensar o problema e achar soluções para resolver ele então eu acredito que o back end seja uma opção para se tornar o foco principal dos meus estudos, vou buscar o conhecimento teórico de ambas linguagem que vc comentou e fazer pesquisas profundas para começar a montar meu portifólio se possível gostaria de um auxilio sobre como começar sabe esse é o real problema que eu venho enfrentando e acaba gerando uma bola de neve que me faz ficar perdido nos estudos mas se vc tiver alguma recomendação de tipo de estudo e qual linguagem centralizar para ser um back end, muito obrigado pela sua ajuda me foi muito utíl para destinguir o caminho certo para eu começar meus trabalhos e estudos

**Reply by davevad93** [reply]
Personally I'd suggest you to start with Python and here's why: first of all it's very beginner friendly and a language that compared to others is very easy to learn, it's also one of the most used languages that has been booming since a few years because as it's the fastest-growing programming language as of today, it's cross-platform which means that you can execute almost any Python program on Windows, Mac, Linux hardware, and even Android and iOS, it's very versatile and suitable for any kind of task, from very simple scripts to more complex programs. Along with Python I'd suggest you to learn Java, they are quite different languages but it's very used and gives a lot of job opportunities. There are many more languages like C# (from Microsoft) C++, Rust, Ruby, Golang et cetera, but I advise you to focus first on Python and Java, if you master those 2 languages you will gain a lot of knowledge and will have many job opportunities in the future.

**Reply by Mathues** [reply]
ok muito obrigado pela ajuda vc me salvou muito espero poder lhe mostrar meus primeiros projetos  após eu dominar essas primeiras linguagens

**Reply by davevad93** [reply]
You're welcome @Mathues , I'm happy to help. [Explore](https://github.com/explore) some repos here on GitHub and take a look at my profile if you want, any other questions you may have, please feel free to ask.
